File: tasks/money-rollup/typescript/environment/src/rollup.ts
export type Item = readonly [account: string, minor: bigint];

export function rollup(items: readonly Item[]): Item[] {
  const totals = new Map<string, bigint>();
  for (const [account, minor] of items) {
    totals.set(account, (totals.get(account) ?? 0n) + minor);
  }
  return [...totals.entries()].sort(([left], [right]) =>
    left < right ? -1 : left > right ? 1 : 0,
  );
}

File: tasks/money-rollup/python-typed/src/rollup.py
"""Aggregate converted entries by account."""

Item = tuple[str, int]


def rollup(items: list[Item]) -> list[Item]:
    """Total each account named by an entry."""
    totals: dict[str, int] = {}
    for account, minor in items:
        totals[account] = totals.get(account, 0) + minor
    return sorted(totals.items())
